core/amber/src/main/scala/edu/uci/ics/amber/error/ErrorLogger.scala

```diff
@@ -0,0 +1,20 @@
+package edu.uci.ics.amber.error
+
+import akka.actor.ActorRef
+import com.typesafe.scalalogging.Logger
+import edu.uci.ics.amber.engine.common.amberexception.WorkflowRuntimeException
+import edu.uci.ics.amber.engine.common.ambermessage.ControlMessage.LogErrorToFrontEnd
+
+object ErrorLogger {
+  private val logger = Logger("WorkflowErrorLogger")
+
+  def logToConsole(err: WorkflowRuntimeError): Unit = {
+    logger.error(err.convertToMap().mkString(" | "))
+  }
+
+  // Use this in all actors except Controller. In Controller, directly call the
+  // eventListener.workflowExecutionErrorListener and pass the error to it.
+  def sendErrToFrontend(controllerRef: ActorRef, err: WorkflowRuntimeError): Unit = {
```

Comment:
So right now our code has to call `sendErrToFrontend` or `logToConsole` in order to do different things. If we look at the logging frameworks, they simply have a `.log()` function, and then externally we can specify where the log will go, like to a file, or to console, or to an event listener, etc..  or even to all of them at the same time.

Can we do something similar here? Let the error logger only expose an `logError` method, and then multiple handlers can handle where to output the error, and we can configure the handlers.